pkg/tree/types.go

```diff
@@ -0,0 +1,34 @@
+package tree
+
+import (
+	"context"
+	"github.com/ipfs/go-cid"
+	"github.com/ipld/go-ipld-prime"
+	"github.com/ipld/go-ipld-prime/linking/cid"
+	"github.com/multiformats/go-multicodec"
+)
+
+// todo: maybe we need clean orphan nodes regularly?
+type NodeStore interface {
+	WriteNode(ctx context.Context, nd *ProllyNode, prefix *cid.Prefix) (cid.Cid, error)
+	ReadNode(ctx context.Context, c cid.Cid) (*ProllyNode, error)
+
+	WriteTree(ctx context.Context, tree *ProllyTree, prefix *cid.Prefix) (cid.Cid, error)
+	ReadTree(ctx context.Context, c cid.Cid) (*ProllyTree, error)
+
+	WriteTreeConfig(ctx context.Context, cfg *TreeConfig, prefix *cid.Prefix) (cid.Cid, error)
+	ReadTreeConfig(ctx context.Context, c cid.Cid) (*TreeConfig, error)
+
+	LinkSystem() *ipld.LinkSystem
+
+	Close()
+}
+
+var DefaultLinkProto = cidlink.LinkPrototype{
+	Prefix: cid.Prefix{
+		Version:  1,
+		Codec:    uint64(multicodec.DagCbor),
+		MhType:   uint64(multicodec.Sha2_256),
+		MhLength: 16,
```

Comment:
Why setting digest length to 16 bytes instead of the default 32? Seems to be pretty uncommon in the IPFS ecosystem.

Reply:
I think to save on overall bytes and to fit more CIDs/KVs into a node.

Is the risk of collision too high with 16?